networkx/algorithms/assortativity/connectivity.py

```diff
@@ -72,9 +75,9 @@ def average_degree_connectivity(
     # First, determine the type of neighbors and the type of degree to use.
     if G.is_directed():
         if source not in ("in", "out", "in+out"):
-            raise ValueError('source must be one of "in", "out", or "in+out"')
+            raise nx.NetworkXError('source must be one of "in", "out", or "in+out"')
```

Comment:
Technically changing the exception type here might bite users who are catching `ValueError`s - is there a strong motivation for doing so?

Reply:
Ah I see - it looks like the goal is to match the exception types in `neighbor_degree`. This seems fine to me then: +1 for consistency and one of the exception types will have to change so might as well be this one!